variables.tf

```diff
@@ -2,8 +2,72 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
-#  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
-#}
+variable "watsonx_data_name" {
+  type        = string
+  description = "The name of the watsonx.data instance."
+  default     = null
+
+  validation {
+    condition     = var.existing_watsonx_data_instance_crn == null ? length(var.watsonx_data_name) > 0 : true
+    error_message = "You must specify a value for 'watsonx_data_name' if 'existing_watsonx_data_instance_crn' is null."
+  }
+}
+
+variable "resource_group_id" {
+  type        = string
+  description = "The resource group ID where the watsonx data instance is created."
+  default     = null
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to created resources"
+  default     = []
+}
+
+variable "region" {
+  type        = string
+  description = "The region to provision the watsonx data instance. "
+  default     = "us-south"
+  validation {
+    condition     = contains(["eu-de", "us-south", "eu-gb", "jp-tok", "au-syd"], var.region)
+    error_message = "You must specify 'eu-de', 'eu-gb', 'jp-tok', 'au-syd' or 'us-south' as the IBM Cloud region."
+  }
+}
+
+variable "existing_watsonx_data_instance_crn" {
+  type        = string
+  description = "The CRN of the an existing watsonx.data instance. If no value is passed, and new instance will be provisioned."
+  default     = null
+
+  validation {
+    condition     = var.existing_watsonx_data_instance_crn != null || var.resource_group_id != null
+    error_message = "When 'existing_instance_crn' is specified then 'resource_group_id' is not required."
```

Comment:
We don't need this kind of validation. Infact we probably don't wan't it either as it would break in a stack that supports passing existing instance.